target_chains/ethereum/contracts/contracts/pulse/scheduler/Scheduler.sol

```diff
@@ -385,6 +390,103 @@ abstract contract Scheduler is IScheduler, SchedulerState {
+        }
+
+        // Check that all existing price IDs are preserved (adding is allowed, not removing)
+        for (uint i = 0; i < currentParams.priceIds.length; i++) {
+            bool found = false;
+            for (uint j = 0; j < newParams.priceIds.length; j++) {
+                if (currentParams.priceIds[i] == newParams.priceIds[j]) {
+                    found = true;
+                    break;
+                }
+            }
+            if (!found) {
+                revert IllegalPermanentSubscriptionModification();
+            }
+        }
+
+        // Cannot change reader whitelist settings for permanent subscriptions
+        if (newParams.whitelistEnabled != currentParams.whitelistEnabled) {
+            revert IllegalPermanentSubscriptionModification();
+        }
+
+        // Check if the set of addresses in the whitelist is the same
+        if (
+            newParams.readerWhitelist.length !=
+            currentParams.readerWhitelist.length
+        ) {
+            revert IllegalPermanentSubscriptionModification();
+        }
+        uint256 n = newParams.readerWhitelist.length;
+        bool[] memory currentVisited = new bool[](n);
+        uint256 matchesFound = 0;
+        for (uint256 i = 0; i < n; i++) {
+            bool foundInCurrent = false;
+            for (uint256 j = 0; j < n; j++) {
+                if (
+                    !currentVisited[j] &&
+                    newParams.readerWhitelist[i] ==
+                    currentParams.readerWhitelist[j]
+                ) {
+                    currentVisited[j] = true;
+                    foundInCurrent = true;
+                    matchesFound++;
+                    break;
+                }
+            }
+            if (!foundInCurrent) {
+                revert IllegalPermanentSubscriptionModification();
+            }
+        }
+
+        // Cannot change update criteria for permanent subscriptions
+        if (
+            newParams.updateCriteria.updateOnHeartbeat !=
+            currentParams.updateCriteria.updateOnHeartbeat ||
+            newParams.updateCriteria.heartbeatSeconds !=
+            currentParams.updateCriteria.heartbeatSeconds ||
+            newParams.updateCriteria.updateOnDeviation !=
+            currentParams.updateCriteria.updateOnDeviation ||
+            newParams.updateCriteria.deviationThresholdBps !=
+            currentParams.updateCriteria.deviationThresholdBps
+        ) {
+            revert IllegalPermanentSubscriptionModification();
+        }
+
+        // Cannot change gas config for permanent subscriptions
+        if (
+            newParams.gasConfig.maxBaseFeeMultiplierCapPct !=
+            currentParams.gasConfig.maxBaseFeeMultiplierCapPct ||
+            newParams.gasConfig.maxPriorityFeeMultiplierCapPct !=
+            currentParams.gasConfig.maxPriorityFeeMultiplierCapPct
+        ) {
+            revert IllegalPermanentSubscriptionModification();
+        }
```

Comment:
I think these gas parameters could cause some problems still. The chain usage will change over time, and these parameters may need to be tweaked. If you can't change them, then they could initially be set too low, and then the chain gets more congested and then you don't get updates.

Maybe we need to set these to infinity for the permanent subscriptions? not sure exactly what to do.

(this is a problem we can resolve later also, so I'm approving this PR)